docs/docs/guides/evaluation/scorers.md

```diff
@@ -0,0 +1,318 @@
+
+**Notes:**
+- Requires the `openai` Python package.
+- The client must be an instance of OpenAI's `OpenAI` or `AsyncOpenAI` client.
+
+---
+
+### `EmbeddingSimilarityScorer`
+
+The `EmbeddingSimilarityScorer` computes the cosine similarity between the embeddings of the AI system's output and a target text from your dataset. It's useful for measuring how similar the AI's output is to a reference text.
+
+```python
+from weave.flow.scorers.similarity_score import EmbeddingSimilarityScorer
+
+llm_client = ...  # initialise your LlM client
+
+similarity_scorer = EmbeddingSimilarityScorer(
+    client=llm_client
+    target_column="reference_text",  # the dataset column to compare the output against
+    threshold=0.4  # the cosine similarity threshold to use  
+)
+```
+
+**Parameters:**
+
+- `target_column`: Name of the dataset column containing the reference text (default is `"text"`).
+- `threshold` (float): Minimum cosine similarity score considered as similar (default is `0.5`).
+
+---
+
+### `ValidJSONScorer`
+
+The ValidJSONScorer checks whether the AI system's output is valid JSON. This scorer is useful when you expect the output to be in JSON format and need to verify its validity.
+
+```python
+from weave.flow.scorers.json_scorer import ValidJSONScorer
+
+json_scorer = ValidJSONScorer()
+```
+
+**Notes:**
+- If the output cannot be parsed as JSON, or if it parses to a data type other than dict or list, it is considered invalid.
+
+
+---
+
+### `ValidXMLScorer`
+
+The `ValidXMLScorer` checks whether the AI system's output is valid XML. This is useful when expecting XML-formatted outputs.
+
+```python
+from weave.flow.scorers.xml_scorer import ValidXMLScorer
+
+xml_scorer = ValidXMLScorer()
+```
+
+---
+
+### `PydanticScorer`
+
+The `PydanticScorer` validates the AI system's output against a Pydantic model to ensure it adheres to a specified schema or data structure.
+
+```python
+from weave.flow.scorers.pydantic_scorer import PydanticScorer
+from pydantic import BaseModel
+
+class FinancialReport(BaseModel):
+    revenue: int
+    year: str
+
+pydantic_scorer = PydanticScorer(model=Person)
+```
+
+---
+
+### RAGAS - `ContextEntityRecallScorer`
+
+The `ContextEntityRecallScorer` estimates context recall by extracting entities from both the AI system's output and the provided context, then computing the recall score. Based on the [RAGAS](https://github.com/explodinggradients/ragas) evaluation library
+
+```python
+from weave.flow.scorers.ragas_scorer import ContextEntityRecallScorer
+
+llm_client = ...  # initialise your LlM client
+
+entity_recall_scorer = ContextEntityRecallScorer(
+    client=llm_client
+    model_id="your-model-id"
+    )
+```
+
+**How It Works:**
+
+- Uses an LLM to extract unique entities from the output and context.
+- Calculates recall as the proportion of entities in the output that are present in the context.
```

Comment:
May benefit from more explanation -- not everyone will know what recall is or how to interpret/set it

Reply:
adding 1 more line of explanation